client.go

```diff
@@ -204,8 +204,8 @@ func (c *Client) cloneClientRequestModifiers() requestModifiers {
 
 	var clone requestModifiers
 
-	copy(clone.requestCallbacks, c.clientRequestModifiers.requestCallbacks)
-	copy(clone.responseCallbacks, c.clientRequestModifiers.responseCallbacks)
+	clone.requestCallbacks = append([]RequestCallback{}, c.clientRequestModifiers.requestCallbacks...)
+	clone.responseCallbacks = append([]ResponseCallback{}, c.clientRequestModifiers.responseCallbacks...)
```

Comment:
don't think ive seen this pattern in golang before, can we use something like https://pkg.go.dev/slices#Clone instead?

Reply:
Nice, this actually [implements](https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/slices/slices.go;l=336-344) what I was trying to do! I definitely like this better, thanks!

Reply:
https://github.com/hashicorp/vault-client-go/pull/242/commits/c439fb70447fb743fb3cef541d648bd66bf69c39